src/Microsoft.DotNet.Build.CloudTestTasks/DownloadFromAzure.cs

```diff
@@ -75,59 +79,87 @@ public override bool Execute()
                         XmlDocument responseFile;
                         using (HttpResponseMessage response = await client.SendAsync(request))
                         {
-                            responseFile = new XmlDocument();
-                            responseFile.LoadXml(await response.Content.ReadAsStringAsync());
-                            XmlNodeList elemList = responseFile.GetElementsByTagName("Name");
+                            if (response.IsSuccessStatusCode)
```

Comment:
I think this is easier to read if it's "if (! successful) { LogError; return false}"  We're gettings SO indented here that it's hard to read the code, because you only have ~40 characters to work with, everything is flowing into multiple lines.

It might also indicate that some refactoring needs to happen (usually if you end up nested more than 2-3 levels, humans trying to read the code can't remember what's happening the all the various up-level scopes)